### PR TITLE
feat: add a `displayName` property to drive objects

### PR DESCRIPTION
**Pull request by jviotti**
The display name property is equal to the device path in macOS and
GNU/Linux, but its equal to the list of volumes with file-systems on
Windows, for user-friendliness purposes.

Change-Type: minor
Signed-off-by: Juan Cruz Viotti <jv@jviotti.com>

### CODE REVIEW

**Comment by jviotti** [comment]
This is something we were manually adding to the result of `drivelist` on Etcher, and its tied to https://github.com/resin-io/etcher/pull/1630

**Comment by lurch** [comment]
Might be nice if the README could show an example of a disk on Windows with multiple drive-letters per disk? (otherwise the fact that this `.name` property might contain a comma-separated list is entirely unclear)

And I wonder if calling it something like `.display_name` would make it clearer that this property is just for display purposes, and you shouldn't really use it for anything else? (given that it has different meanings on different platforms)

**Comment by jviotti** [comment]
> Might be nice if the README could show an example of a disk on Windows with multiple drive-letters per disk? (otherwise the fact that this .name property might contain a comma-separated list is entirely unclear)

Sounds good. I'll do it in a second.

> And I wonder if calling it something like .display_name would make it clearer that this property is just for display purposes, and you shouldn't really use it for anything else? (given that it has different meanings on different platforms)

Fair enough, lets go for it!

File: src/windows/scanner.cc [hunk 1]
```diff
@@ -245,6 +246,22 @@ drivelist::Scanner::Scan(std::vector<drivelist::disk_s> *const output) {
 
   for (drivelist::disk_s disk : disks) {
     disk.mountpoints = mountpointMap[disk.id];
+
+    // For user friendly purposes, in Windows, the disk name is
+    // simply the list of drive letters, separated by commas.
+    disk.displayName = std::accumulate(
```
Comment (jviotti) [inline src/windows/scanner.cc]: Uh, I just realized I'm not handling the case where there are no mountpoints :/ I'll fix it in a minute

Comment (lurch) [inline src/windows/scanner.cc]: Good catch! Could you also add an example of that to the README please?

File: lib/scripts.json [hunk 2]
```diff
@@ -1,12 +1,12 @@
...
     "originalFilename": "linux.sh",
     "type": "text"
   }
-}
+}
```
Comment (jhermsmeier) [inline lib/scripts.json]: Missing EOL here

File: lib/parse.js [hunk 3]
```diff
@@ -68,7 +68,7 @@ module.exports = (input) => {
     device = _.chain(device)
       .split('\n')
       .filter((line) => {
-        return /^(\s\s-\s)?[a-z]+:/g.test(line);
+        return /^(\s\s-\s)?[a-zA-Z]+:/g.test(line);
```
Comment (jhermsmeier) [inline lib/parse.js]: This can be changed to `/^(\s{2}-\s)?[a-z]/i` - no need for a global match flag when testing for the start of a string

File: src/windows/scanner.cc (hunk 1 above)
Comment (jviotti) [inline src/windows/scanner.cc]: Done!

File: lib/scripts.json (hunk 2 above)
Comment (jviotti) [inline lib/scripts.json]: Fixed!

File: lib/parse.js (hunk 3 above)
Comment (jviotti) [inline lib/parse.js]: Good catch! 

> Also, why is this testing for \s\s-\s, when none of the example input has that prefix?
Is that something that one of the scripts produces?

Mountpoints have that prefix. For example:

```yaml
device: /dev/disk1
displayName: /dev/disk1
description: "Macintosh HD - APPLE SSD SM0256G"
size: 140999917568
mountpoints:
  - path: "/"
raw: /dev/rdisk1
protected: False
system: True
```

Comment (jhermsmeier) [inline lib/parse.js]: Ah, missed that

File: lib/scripts.json (hunk 2 above)
Comment (lurch) [inline lib/scripts.json]: Given that this file is auto-generated anyway, how did this 'mistake' creep in?

Comment (jviotti) [inline lib/scripts.json]: Generating the file from Windows messed things up :(
